app/src/main/java/com/example/pygmyhippo/organizer/PostEventFragment.java

```diff
@@ -204,36 +199,75 @@ public void clearEditTextFields(){
         if (queryID == 0) {
+            if (flags == DBOnCompleteFlags.SUCCESS.value) {
+                Event newEvent = docs.get(0);
+                if(newEvent.getHashcode() == 0){
+                    boolean hashcodeValid = newEvent.generateHashcode();
+                    if(hashcodeValid) {
+                        Log.i("Post Event", String.format("Event generated with hashcode", newEvent.getHashcode()));
+                        handler.updateEvent(newEvent, this);
+                    }
+                    else {
+                        Log.e("Post Event", "Hashcode generation unsuccessful");
+                        Toast.makeText(getContext(), "Event Failed to Create", Toast.LENGTH_LONG).show();
+                    }
+                }
+            } else {
+                Log.e("Post Event", "Hashcode generation unsuccessful");
+                Toast.makeText(getContext(), "Event Failed to Create", Toast.LENGTH_LONG).show();
+            }
+        }
+        else if (queryID == 2){
             if (flags == DBOnCompleteFlags.SUCCESS.value) {
                 clearEditTextFields();
                 Event newEvent = docs.get(0);
-                // create a bundle to send an event to the qr code widget
-                // the bundle must have a id for the qr code to generate
-                // event should be added to data base then it should be
                 Bundle navArgs = new Bundle();
                 navArgs.putString("eventID", newEvent.getEventID());
                 navArgs.putParcelable("signedInAccount", signedInAccount);
                 navController.navigate(R.id.view_eventqr_fragment, navArgs);
-            } else {
+            }
+            else {
+                Log.e("Post Event", "Hashcode generation unsuccessful");
                 Toast.makeText(getContext(), "Event Failed to Create", Toast.LENGTH_LONG).show();
             }
         }
-        else if (queryID == 3){
+    }
+
+    /**
+     * Callback called when view entrant Storage queries complete.
+     * @param docs - Documents retrieved from DB (if it was a get query).
+     * @param queryID - ID of query completed.
+     * @param flags - Flags to indicate query status/set how to process query result.
+     */
+    @Override
+    public void OnCompleteStorage(@NonNull ArrayList<Image> docs, int queryID, int flags) {
+          if (queryID == 1){
             if (flags == DBOnCompleteFlags.SUCCESS.value) {
+                Log.i("Post Event", "Image upload successful");
                 clearEditTextFields();
-                Event newEvent = docs.get(0);
-                imagePath = newEvent.getEventPoster();
+                Image newImage = docs.get(0);
+                imagePath = newImage.getUrl(); // set the image path from the URL
             }
             else {
+                Log.e("Post Event", "Image upload unsuccessful");
                 Toast.makeText(getContext(), "Image failed to upload", Toast.LENGTH_LONG).show();
```

Comment:
Do we still have the whole new event being added to the database? I don't see it right off the bat... Or I'm probably just blind

Reply:
I don't really get what you're asking. How events relate to images, or how the events are uploaded?

Reply:
Kinda both. I just want to make sure I'm understanding how its stored, are events stored separately from their poster images?